packages/angular_devkit/build_angular/src/babel/webpack-loader.ts

```diff
@@ -0,0 +1,83 @@
+/**
+ * @license
+ * Copyright Google Inc. All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+import { VERSION } from '@angular/compiler-cli';
+import { custom } from 'babel-loader';
+
+interface AngularCustomOptions {
+  forceES5: boolean;
+  requiresLinking: boolean;
+}
+
+export default custom<AngularCustomOptions>(() => {
+  const baseOptions = Object.freeze({
+    babelrc: false,
+    configFile: false,
+    compact: false,
+    cacheCompression: false,
+    sourceType: 'unambiguous',
+  });
+  let hasLinkerSupport: undefined | boolean;
+
+  return {
+    customOptions({ forceES5, ...loaderOptions }, { source }) {
+      const requiresLinking =
+        (source.includes('ɵɵngDeclareDirective') || source.includes('ɵɵngDeclareComponent')) &&
+        !source.includes('function ɵɵngDeclareDirective(decl)');
```

Comment:
Now that https://github.com/angular/angular/pull/40137 has landed, we can use the provided function.